### DISCUSSION TITLE
Correct method to use fs.ReadStream to create message from file stream

### DISCUSSION THREAD

**Discussion by eXhumer** [openpgpjs/openpgpjs#1634] [Q&A]
I want to stream a large file and verify it with a key. However, using `fs.createReadStream("<FILE-PATH>")` doesn't seem to work. There is no console.log output on the console.

```
async function verifyFileSignature(fileName) {
  const result = await openpgp.verify({
    message: await openpgp.createMessage({
      binary: fs.createReadStream(path.join(__dirname, fileName)),
      format: "binary",
    }),
    signature: await openpgp.readSignature({
      binarySignature: fs.readFileSync(path.join(__dirname, `${fileName}.sig`)),
    }),
    verificationKeys: await openpgp.readKey({ armoredKey: armoredKey }),
    expectSigned: true,
  });

  for await (const chunk of result.data) {}

  const verified = await result.signatures[0].verified;
  console.log(`Verified: ${verified}`);
}

openpgp.config.allowInsecureVerificationWithReformattedKeys = true;
openpgp.config.allowUnauthenticatedMessages = true;
openpgp.config.allowUnauthenticatedStream = true;
verifyFileSignature("eXViewer-win32-x64-1.0.0.zip");
```

Output
```
PS C:\Users\eXhumer\Projects\openpgp-test> yarn start
yarn run v1.22.19
$ node .
Done in 0.30s.
```

Reading the whole file with `fs.readFileSync(path.join(__dirname, fileName))` and then passing that buffer to `binary` parameter works, but that requires me to read the whole file first in memory to do so. I would like to avoid reading the whole file, if possible.

I [found a discussion that uses `fs.ReadStream`](https://github.com/openpgpjs/openpgpjs/discussions/1305), but doing the resolutions mentioned doesn't fix the problem on my end.

**Comment by larabr** [comment]
Hi @eXhumer, note that in the provided code it seems you added the `format: 'binary'` option to `createMessage` instead of `verify`. That said, this should not result in the issue you are describing. Can you add more log statements to see where is it that the code exits? 
Also, what Node version are you using?

**Reply by mhassan1** [reply]
This appears to be a similar question to https://github.com/openpgpjs/openpgpjs/discussions/1707.
